enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -591,10 +592,29 @@ def can_redeem(self, lms_user_id, content_key, skip_customer_user_check=False):
 
         # learner not associated to enterprise
         if not skip_customer_user_check:
-            if self.lms_api_client.get_enterprise_user(self.enterprise_customer_uuid, lms_user_id) is None:
+            learner_record = self.lms_api_client.get_enterprise_user(self.enterprise_customer_uuid, lms_user_id)
+            if not learner_record:
                 self._log_redeemability(False, REASON_LEARNER_NOT_IN_ENTERPRISE, lms_user_id, content_key)
                 return (False, REASON_LEARNER_NOT_IN_ENTERPRISE, [])
 
+            associated_group_uuids = set(  # pylint: disable=consider-using-set-comprehension
+                [str(group.get('uuid')) for group in learner_record.get('enterprise_group')]
```

Comment:
With the release of https://github.com/openedx/edx-enterprise/pull/2040 we'll have to update this logic to account for the new group data structure in the enterprise customer serializer